doc/nrf/ug_creating_matter_accessory.rst

```diff
@@ -0,0 +1,274 @@
+
+
+| To edit :file:`template.zap` run the command below with ``MATTER_ROOT`` replaced by a path to the root of Matter repository:
+.. code-block:: 
+
+        $MATTER_ROOT/scripts/tools/zap/run_zaptool.sh src/template.zap 
+        
+
+| First, add a new endpoint representing a sensor device:
+.. image:: images/create_matter_accessory_create_new_endpoint.png
+
+|
+| Only two clusters are needed in that endpoint:
+
+#. OnOff server.
+#. TemperatureMeasurement server.
+
+| First one can be added in General category:
+.. image:: images/create_matter_accessory_add_onoff_cluster.png
+
+|
+| Second one can be found in Measurement & Sensing:
+.. image:: images/create_matter_accessory_add_temperature_measurement.png
+
+|
+| Save the file and exit.
+
+| Created ZAP file will be used to generate C++ code containing selected clusters.
+| In order to generate this code in :file:`src/gen/` run: 
+.. code-block:: 
+        
+        $MATTER_ROOT/scripts/tools/zap/generate.py src/template.zap
+       
+At this point, new clusters have been added to the Matter accessory. 
+
+Main application
+****************
+
+| Adding clusters alone is not enough to properly model sensor's behaviour.
+| Next step in creating a sensor is changing a way in which an application will interact with newly added clusters. 
+| The easiest way to do it is to modify :file:`src/app_task.cpp`, which contains main loop of the application.
+
+
+Event queue
+===========
+
+| Main application loop is based on an queue on which events are posted by ZCL callbacks, application itself or other entities, like Zephyr timers.
+| In each iteration an event is dequeued and a corresponding event handler is called.
+
+Events
+------
+
+| In a template sample application events are representing:
+
+#. Pressing Button 1.
+#. Releasing Button 1.
+#. Factory reset timeout.
+
+To model behaviour of a sensor you need to add three new events:
+
+#. Sensor activation.
+#. Sensor deactivation.
+#. Sensor measurement.
+
+| Add new events to ``EventType`` in :file:`src/app_event.h`, for example:
+.. code-block:: C++
+        
+        enum EventType : uint8_t { FunctionPress, FunctionRelease, FunctionTimer, SensorActivate, SensorDeactivate, SensorMeasure};
+
+Timers
+------
+
+| One way to make sure that sensor is making measurements in desired time points is to use a Zephyr timer to post ``SensorMeasure`` events.
+| In Template sample such a timer is being used in order to countdown 6 seconds of **Button 1** being pressed to initiate factory reset.
+
+New timer for a measurement event can be added in a similar way:
+
+.. code-block:: C++
+
+        k_timer sSensorTimer;        
+
+        void SensorTimerHandler(k_timer *timer)
+        {
+                GetAppTask().PostEvent(AppEvent{ AppEvent::SensorMeasure });
+        }
+        
+        void StartSensorTimer(uint32_t aTimeoutMs)
+        {
+	        k_timer_start(&sSensorTimer, K_MSEC(aTimeoutMs), K_NO_WAIT);
```

Comment:
```suggestion
	        k_timer_start(&sSensorTimer, K_MSEC(aTimeoutMs), K_MSEC(aTimeoutMs));
```
We don't want the timer to be one-shot. Then you wouldn't need to re-start it on each timeout.